src/apps/competitions/models.py

```diff
@@ -23,6 +28,107 @@ def __str__(self):
     def bundle_dataset(self):
         return CompetitionCreationTaskStatus.objects.get(resulting_competition=self).dataset
 
+    def check_future_phase_submissions(self):
+        """
+        Checks for if we need to migrate current phase submissions to next phase.
+        """
+        phases = self.phases.all()
+
+        if len(phases) == 0:
+            return
+
+        try:
+            current_phase = phases.get(status='Current')
+            next_phase = phases.get(status='Next')
+            final_phase = phases.get(status='Final')
+
+        except Phase.DoesNotExist:
+            return
+
+        # Making sure current_phase or next_phase is not None
+        if current_phase is None or next_phase is None or current_phase is final_phase:
+            return
+
+        logger.info("Checking for needed migrations on competition pk=%s, current phase: %s, next phase: %s" %
+                    (self.pk, current_phase.index, next_phase.index))
+
+        # Check for next phase and see if it has auto_migration enabled
+        if next_phase and next_phase.auto_migration:
+            if next_phase.start < now() and current_phase.end < now():
```

Comment:
Probably also `and current_phase.auto_migration` so we don't migrate if we aren't supposed to.